### PR TITLE
Cleanup - Handle function errors, remove unnecessary assignment, pass arguments more efficiently

### PR DESCRIPTION
**Pull request by trimental**
I think its better to panic on functions that return results even if it doesn't matter too much if the function fails, or at least they should be printed to stderr to log the failure.

A few arguments can be passed more efficiently, proxies by reference probably don't save a lot performance wise but are more idiomatic and now's probably a good time to change these considering the next release will be breaking.

### CODE REVIEW

**Comment by vberger** [comment]
Sorry for the delay on coming back on this.

For errors that are non-fatal (for example failing to draw, it *could* be an intermittent error ?), I'd rather just print something on stderr and continue, rather than panic.

File: src/window/basic_frame.rs [hunk 1]
```diff
@@ -288,14 +288,12 @@ impl Frame for BasicFrame {
...
-                            change_pointer(&pointer, data.location, Some(serial));
+                            change_pointer(&pointer, data.location, Some(serial))
                         }
                     }
                     Event::Leave { serial, .. } => {
                         data.location = Location::None;
-                        if resizable {
-                            change_pointer(&pointer, data.location, Some(serial));
-                        }
+                        change_pointer(&pointer, data.location, Some(serial))
```
Comment (vberger) [inline src/window/basic_frame.rs]: Why was this `if` removed ?

Comment (trimental) [reply src/window/basic_frame.rs]: Ah, to be honest I think this is just a mistake I didn't catch in the git diff

**Comment by vberger** [comment]
Regarding printing messages to stderr, I tend to prefix the messages according to the crate generating them, and add some information about where the error originated. So something like:

`[SCTK] Data offer: failed to close write pipe: {:?}`

**Pull request merged**